client/packages/common/src/ui/forms/JsonForms/components/Array.tsx

```diff
@@ -0,0 +1,189 @@
+  JsonFormsDispatch,
+} from '@jsonforms/react';
+import {
+  Box,
+  Typography,
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+} from '@mui/material';
+import {
+  IconButton,
+  PlusCircleIcon,
+  MinusCircleIcon,
+  ChevronDownIcon,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { RegexUtils } from '@common/utils';
+import {
+  FORM_LABEL_COLUMN_WIDTH,
+  FORM_INPUT_COLUMN_WIDTH,
+} from '../styleConstants';
+import { JsonData } from '../useJsonForms';
+
+interface UISchemaWithCustomProps extends ControlElement {
+  defaultNewItem?: JsonData;
+  itemLabel?: string;
+}
+
+interface ArrayControlCustomProps extends ArrayControlProps {
+  uischema: UISchemaWithCustomProps;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+  data: JsonData[];
+}
+
+interface FoundUiSchema extends Layout {
+  elements: ControlElement[];
+}
+
+export const arrayTester = rankWith(5, schemaTypeIs('array'));
+
+const UIComponent = (props: ArrayControlCustomProps) => {
+  const t = useTranslation('common');
+  const {
+    uischema,
+    uischemas,
+    schema,
+    path,
+    data,
+    addItem,
+    removeItems,
+    enabled,
+    label,
+    rootSchema,
+  } = props;
+
+  const foundUISchema = useMemo(
+    () =>
+      findUISchema(
+        uischemas as JsonFormsUISchemaRegistryEntry[],
+        schema,
+        uischema.scope,
+        path,
+        undefined,
+        uischema,
+        rootSchema
+      ),
+    []
+  ) as FoundUiSchema;
+
+  return (
+    <Box display="flex" flexDirection="column" gap={0.5} marginTop={2}>
+      <Box display="flex" width="100%" gap={2} alignItems="center">
+        <Box width={FORM_LABEL_COLUMN_WIDTH}>
+          <Typography sx={{ fontWeight: 'bold', textAlign: 'end' }}>
+            {label}:
+          </Typography>
+        </Box>
+        <Box width={FORM_INPUT_COLUMN_WIDTH} textAlign="right">
+          <IconButton
+            icon={<PlusCircleIcon />}
+            label={t('label.add-another')}
+            color="primary"
+            onClick={addItem(
+              path,
+              uischema.defaultNewItem ?? createNewItem(foundUISchema)
+            )}
+          />
+        </Box>
+      </Box>
+      {data.map((child, index) => {
+        return (
+          <Accordion
+            key={index}
+            defaultExpanded={index === data.length - 1}
+            sx={{
+              mt: '0 !important',
+              mb: index === data.length - 1 ? '20px !important' : 1,
+            }}
+          >
+            <AccordionSummary
+              expandIcon={<ChevronDownIcon />}
+              sx={{
+                '&:hover .array-remove-icon': { visibility: 'visible' },
+                margin: '0 !important',
+                minHeight: '0 !important',
+                '.MuiAccordionSummary-content': {
+                  margin: '5px !important',
+                },
+                '.Mui-expanded': {
+                  marginBottom: '0 !important',
```

Comment:
Don't love all these `!important`s, but they don't make it easy to over-ride the internal components of Accordion!

Reply:
Oh right -- so "style" gets rendered inline and therefore takes precedence, but "sx" goes to the component class, is that right?

Cool, I'll play round with that and see what works best. Thanks

Reply:
Yeah, the "style" prop can only take know css properties as you suspected, so only marginal improvement: a903e0c46

Reply:
yup, that's right. style becomes inline style on the element, `sx` is some material-ui magic